surface index/inner index

tarikceric opened the pull request:
Surface index/inner_index in active silver `nft_sales_xxx` tables. Tests added for new columns and recency filter.

To update: Locally FR'ed tables in DEV, and to speed it up I added `and tx_id in (select tx_id from {{ this }})` in certain CTE's referencing transfers/decoded_instructions/etc. I will then use an adhoc `update` statement on index + inner_index cols from DEV to PROD

- incremental runs on medium
19:17:11  4 of 7 OK created sql incremental model silver.nft_sales_solsniper ............. [SUCCESS 1 in 17.73s]
19:17:12  2 of 7 OK created sql incremental model silver.nft_sales_magic_eden_cnft_onchain  [SUCCESS 8 in 18.70s]
19:17:22  1 of 7 OK created sql incremental model silver.nft_sales_exchange_art .......... [SUCCESS 1 in 28.71s]
19:17:30  5 of 7 OK created sql incremental model silver.nft_sales_solsniper_cnft_onchain  [SUCCESS 2 in 19.04s]
19:17:31  7 of 7 OK created sql incremental model silver.nft_sales_magic_eden_cnft ....... [SUCCESS 460 in 9.35s]
19:17:33  6 of 7 OK created sql incremental model silver.nft_sales_tensorswap_cnft ....... [SUCCESS 55 in 21.36s]
19:17:34  3 of 7 OK created sql incremental model silver.nft_sales_solanart .............. [SUCCESS 1 in 41.03s]

- test (5 errors are from recency test as dev isnt refreshed)
`19:34:35  Done. PASS=103 WARN=0 ERROR=5 SKIP=0 TOTAL=108`

desmond-hui reviewed (commented):
> to speed it up I added and tx_id in (select tx_id from {{ this }}) in certain CTE's referencing transfers/decoded_instructions/etc.

I don't see any changes w/ this in place. Also I think there are models missing that should be included in this update like `silver__nft_sales_magic_eden_v2`?

tarikceric commented:
> > to speed it up I added and tx_id in (select tx_id from {{ this }}) in certain CTE's referencing transfers/decoded_instructions/etc.
> 
> I don't see any changes w/ this in place?

it was just a note on what i added ad-hoc locally to make the one-time FR run faster since there arent any core logic changes, but wont be useful to add for future FR's when there is a logic change that would return other tx's

desmond-hui reviewed (approved):
Since you are updating the schema files for all these, can you also change `tests` to `data_tests`? `tests` has been deprecated as of dbt 1.8
